### ISSUE TITLE
Pushd rotation adds new copy of pwd, doesn't rotate

### ISSUE THREAD

**Issue by cmajames** [fish-shell/fish-shell#4398]
Environment: 
`fish --version`: fish, version 2.4.0
`uname -a`: Linux q2 4.10.0-33-generic #37-Ubuntu SMP Fri Aug 11 10:55:28 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
`echo $TERM`: xterm-256color
`sh -c 'env HOME=$(mktemp -d) fish'` doesn't help

Recreation:
1. Add 2+ dirs to the active stack using `pushd <dir>`
2. Attempt to rotate through dirs using `pushd <[+|-]N>`

Exemplar Transcript: 
```shell
cmajames@q2 /h/c/org> dirs
/home/cmajames/org  /home/cmajames  /home/cmajames/Private  
cmajames@q2 /h/c/org> pushd +1
cd: The directory “+1” does not exist
cmajames@q2 /h/c/org> dirs
/home/cmajames/org  /home/cmajames/org  /home/cmajames  /home/cmajames/Private  
cmajames@q2 /h/c/org> pushd -1
cd: The directory “-1” does not exist
cmajames@q2 /h/c/org> dirs
/home/cmajames/org  /home/cmajames/org  /home/cmajames/org  /home/cmajames  /home/cmajames/Private  
cmajames@q2 /h/c/org> 
```

**Comment by zanchey** [comment]
The `+1` case is fixed in current master, but `-1` produces an argparse error.

**Comment by krader1961** [comment]
Mea Culpa! Just revert my commit 05aae4764. The `popd` and `pushd` commands are weird. They don't officially support the `--help` flag (or it variants) even though the implementations do so. And `pushd` supports the `+n` and `-n` syntax to rotate the directory stack in addition to pushing a new directory on the stack. Something I've never been able to wrap my mind around and thus have never been able to use. Which I why I wrote my `mcd` command for ksh and zsh and the `cdh` command for fish. Still, that is no reason to break the legacy behavior of `pushd`. It would be a good idea to include a comment in those functions as to why they don't use `argparse` so that future readers of that code understand why they don't parse their arguments in the normal manner.

**Comment by zanchey** [comment]
I don't think +1 and -1 work properly at all before that commit - certainly I can't make them work in 2.6.0.

**Comment by zanchey** [comment]
Right, #4091 fixed it after 2.6.0.

@cmajames, this is a duplicate of #4091, which is fixed in the upcoming release.

**Comment by cmajames** [comment]
Great, thanks
